### DISCUSSION TITLE
Given a post title, how to get the permalink of the post in javascript code?

### DISCUSSION THREAD

**Discussion by Unparalleled-Calvin** [hexojs/hexo#5286] [Question / Answer]
I am writing my hexo plugin whose tag format looks like {% book_card  [post_title] %}. I am confused about how to get the permalink of the post because I want to generate a hyperlink to my post from the title given by the user.
I have noticed that hexo offers tags like `post_link` to help users convet title to link, but js code can't utilize this gadget. Do you have any ideas ? Any help will be appreciated, thx.

btw, the version I'm using now is 6.3.0.

**Comment by uiolee** [comment]
`posts` data in hexo.locals.posts

https://hexo.io/api/locals

**Reply by Unparalleled-Calvin** [reply]
Thank you for your reply!
```
const log = require('hexo-log')({ 'debug': false, 'slient': false });

...

function tagTemplate(book_name, author_name, read_time, short_comment, pic_path) {
    hexo.locals.get('posts').forEach((v, i) => {
        log.info(v.title);
    })
    tag = `...`
    return tag
}

hexo.extend.tag.register('book_card', function (args) {
    let [book_name, author_name, read_time, short_comment, pic_path] = args
    return tagTemplate(book_name, author_name, read_time, short_comment, pic_path)
})

...
```
Here are a few of lines in my code. When I use `hexo cl && hexo s` to restart the engine, the log only prints **part** of my posts' titles maybe 2 or 3. I guess it is caused by concurrent processing or some processing order. What should I do now?

**Reply by D-Sketon** [reply]
also see #5194, #5230, this may be a bug

**Reply by uiolee** [reply]
i make test and it works well. 

![图片](https://github.com/hexojs/hexo/assets/22849383/55ca983b-090c-4bb2-ae41-d7fcfc09552f)
